packages/namekit-react/src/components/CurrencySymbol.tsx

```diff
@@ -0,0 +1,161 @@
+import { PriceCurrencyFormat, Currency } from "@namehash/ens-utils";
+import { Tooltip } from "./Tooltip";
+
+import { UsdSymbol } from "./icons/UsdSymbol";
+import { UsdcSymbol } from "./icons/UsdcSymbol";
+import { WethSymbol } from "./icons/WethSymbol";
+import { EthSymbol } from "./icons/EthSymbol";
+import { DaiSymbol } from "./icons/DaiSymbol";
+import React from "react";
+
+/**
+ * The size of the `CurrencySymbol`.
+ * This is defined as a mapping from a `CurrencySymbolSize` to a number.
+ * This number represents the size (width and height) a `CurrencySymbol` should be displayed at, in pixels.
+ */
+export const CurrencySymbolSize = {
+  Small: 16,
+  Large: 20,
+} as const;
+export type CurrencySymbolSize =
+  (typeof CurrencySymbolSize)[keyof typeof CurrencySymbolSize];
+
+export const CurrencySymbology = {
+  /**
+   * The symbol displayed for `Currency` will be its acronym as text. (e.g. "USD").
+   */
+  Acronym: "Acronym",
+  /**
+   * The symbol displayed for `Currency` will be its text-based symbol (e.g. "$").
+   */
+  Symbol: "Symbol",
+  /**
+   * The symbol displayed for `Currency` will be its graphical icon (e.g. the Ethereum Icon as an SVG).
+   */
+  Icon: "Icon",
+} as const;
+export type CurrencySymbology =
+  (typeof CurrencySymbology)[keyof typeof CurrencySymbology];
+
+interface CurrencySymbolProps extends React.ComponentProps<React.ElementType> {
+  /**
+   * The `Currency` to display the symbol for.
+   */
+  currency: Currency;
+
+  /**
+   * The size of the `CurrencySymbol` icon.
+   *
+   * Defaults to `CurrencySymbolSize.Small`.
+   * Only applicable when `symbology` is `CurrencySymbology.Icon`.
+   * If you want to set a custom size for other symbologies, use props to define it (e.g. className="myCustomClassForTextSize")
+   */
+  iconSize?: CurrencySymbolSize;
+
+  /**
+   * If `true`, hovering over the `CurrencySymbol` will display the
+   * name of `currency` in a `Tooltip`. If `false` then the `CurrencySymbol`
+   * won't have any `Tooltip`.
+   *
+   * Defaults to `true`.
+   */
+  describeCurrencyInTooltip?: boolean;
+
+  /**
+   * The symbology to use when displaying the currency.
+   *
+   * Defaults to `CurrencySymbology.Symbol`.
+   */
+  symbology?: CurrencySymbology;
+}
+
+/**
+ *
+ * @param currency: Currency. The currency to get the Icon for (e.g. Currency.Eth)
+ * @param iconSize: CurrencySymbolSize. The Icon size (width and height), in pixels.
+ * @returns
+ */
+const getCurrencyIcon = ({
+  currency,
+  iconSize = CurrencySymbolSize.Small,
+  ...props
+}: {
+  iconSize: CurrencySymbolSize;
+  currency: Currency;
+}): JSX.Element => {
+  switch (currency) {
+    case Currency.Usd:
+      return <UsdSymbol {...props} />;
```

Comment:
Suggest making a comment here explaining how for USD we always display it as a text symbol and never as an icon.